### ISSUE TITLE
some problem about cnnseg model inference

### ISSUE THREAD

**Issue by muzi2045** [kosuke55/train_baiducnn#1]
Hi, thanks for your work!
I'm trying to train the cnnseg model with pytorch, and export onnx model from pytorch.
here is the onnx model
[cnnseg_sim.onnx.tar.gz](https://github.com/kosuke55/train_baiducnn/files/4598141/cnnseg_sim.onnx.tar.gz)
and convert it to trt file
[cnnseg.engine.tar.gz](https://github.com/kosuke55/train_baiducnn/files/4598153/cnnseg.engine.tar.gz)
And using the engine file export from trained in [lidar_apollo_instance_segmentation](https://github.com/tier4/AutowareArchitectureProposal/tree/master/src/perception/object_recognition/detection/lidar_apollo_instance_segmentation)

but it looks like has no any output cluster from network inference with my own trained onnx file.
with the same repo, it can got normal result with original engine file.
thanks for any advice.
@kosuke55

**Comment by kosuke55** [comment]
@muzi2045 
Thank you for trying it out. Currently, this repository is still under development and has many problems, so the accuracy is not yet high. It looks like your branch is behind mine. I've recently changed the input and output features, size, etc., so please pull and try the latest version. There are still many changes to come and I apologize for the inconvenience.

**Comment by muzi2045** [comment]
OK , I'll try to pull the latest code and check the output pred tensor in python first.
the input tensor shape is [1, 8, 640, 640], the output tensor shape is [1, 6, 640, 640].
when training, the GT and Pred looks like no problem:
![image](https://user-images.githubusercontent.com/12984898/81400895-b226da00-9160-11ea-8a41-8c6814eff994.png)
so maybe there something wrong in postprocess part or preprocess part.
@kosuke55

**Comment by muzi2045** [comment]
Why the newest repo code in network output tensor is [1, 12 , 640, 640] ?
```
confidence = torch.sigmoid(deconv0[:, 3:4, :, :])
pred_class = F.softmax(deconv0[:, 4:10, :, :])
output = torch.cat(
      [deconv0[:, 0:3, ...], confidence,
      pred_class, deconv0[:, 10:, ...]], dim=1)
```
in the original caffe model, the output is [1, 6, 672, 672]:
![image](https://user-images.githubusercontent.com/12984898/81405035-73e1e880-9169-11ea-95ad-c0693f64ff5c.png)

because the pred tensor need to predict more classes?

**Comment by kosuke55** [comment]
@muzi2045 
The old code only predicted confidence and class. (You can see these outputs in [tensorrt_bcnn](https://github.com/kosuke55/tensorrt_bcnn/blob/9500db93b3e2a305ad54da9dc0b73cca586b53ed/src/tensorrt_bcnn_ros.cpp#L76-L77)'s confidence_image and so on.) In other words, it didn't support the learning of instance_pt and height_pt required for the post-processing phase.
The new code learns these things.
The 6 channel layer you are checking is the [input](https://github.com/tier4/AutowareArchitectureProposal/blob/cc3dee8c0166f7edf521131aae23813070ae393a/src/perception/object_recognition/detection/lidar_apollo_instance_segmentation/data/hdl-64.prototxt#L6-L12) and the shape of the [output](https://github.com/tier4/AutowareArchitectureProposal/blob/cc3dee8c0166f7edf521131aae23813070ae393a/src/perception/object_recognition/detection/lidar_apollo_instance_segmentation/data/hdl-64.prototxt#L673) is correct at 12 channels.

**Comment by muzi2045** [comment]
thanks a lot, I have figure it out.
